pyomo/contrib/doe/fim_doe.py

```diff
@@ -0,0 +1,2292 @@
+                 prior_FIM=None, discretize_model=None, verbose=True, args=None, measurement_extra_index=None):
+        '''
+        This package enables model-based design of experiments analysis with Pyomo. Both direct optimization and enumeration modes are supported.
+        NLP sensitivity tools, e.g.,  sipopt and k_aug, are supported to accelerate analysis via enumeration.
+        It can be applied to dynamic models, where design variables are controlled throughout the experiment.
+
+        Parameters:
+        -----------
+        param_init: a dictionary of parameter names and values. If they are an indexed variable, put the variable name and index, such as 'theta["A1"]'. Note: if sIPOPT is used, parameter shouldn't be indexed. 
+        design_variable_timepoints: a dictionary, keys are design variable names, values are its control time points.
+                if this design var is independent of time (constant), set the time to [0]
+        measurement_variables: the variable name of the model output, for e.g., ['CA', 'CB', 'CC'].
+        measurement_timeset: a list of measurement time points. can be different from control time points
+        create_model: a function that returns the model, where:
+                      - parameter and design variables are defined as variables
+                      - define every state variables dependent on parameters with a scenario index
+                      - take scenarios as the first argument of this function
+                      - define time index as 't'.
+                      - design variables are defined with and only with a time index.
+        solver: User specified solver, default=None. If not specified, default solver is IPOPT MA57.
+        prior_FIM: Fisher information matrix (FIM) for prior experiments, default=None
+        discretize_model: A user-specified function that deiscretizes the model. Only use with Pyomo.DAE, default=None
+        verbose: if print statements are made
+        args: Other arguments of the create_model function, in a list
+        measurement_extra_index: if measurement variables have extra index between scenario and time indexes. A dictionary, keys are the measurement names, values are a list of indexes.
+        '''  
+        
+        # parameters
+        self.param_init = param_init
+        self.param_name = list(param_init.keys())
+        self.param_value = list(param_init.values())
+        # design variable name
+        self.design_timeset = design_variable_timepoints
+        self.design_name = list(self.design_timeset.keys())
+        # the control time point for each design variable
+        self.design_time = list(self.design_timeset.values())
+        # model output (measurement) name
+        self.measurement_variables = measurement_variables
+        # model measurement time
+        self.measurement_timeset = measurement_timeset
```

Comment:
Why is desigin_timeset split into design_time and design_name, but this is not done for the measurement_timeset?